### PR TITLE
`Vector{2,3}.__delattr__` messaging fixed when deleting `x, y, {z}` and removed `Vector4` zombies

### LINKED ISSUE #3066: Incorrect error message for vectors
I was digging through the math.c source and found this:

```py
>>> import pygame
pygame-ce 2.5.2.dev1 (SDL 2.30.6, Python 3.9.5)
>>> a = pygame.Vector2(30,40)
>>> del a.y
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Cannot delete the x attribute
```

I didn't try to delete the x attribute, I tried to delete the y attribute!

Implementation code (unchanged for 8 years):
```c
static int
vector_set_component(pgVector *self, PyObject *value, int component)
{
    if (value == NULL) {
        PyErr_SetString(PyExc_TypeError, "Cannot delete the x attribute");
        return -1;
    }
    if (component >= self->dim) {
        PyErr_BadInternalCall();
        return -1;
    }

    self->coords[component] = PyFloat_AsDouble(value);
    if (PyErr_Occurred())
        return -1;
    return 0;
}
```

### PR DESCRIPTION
**Pull request by oddbookworm**
Fixes #3066 and removes the remnants of whatever Vector4 used to be there

### CODE REVIEW

File: src_c/math.c [hunk 1]
```diff
@@ -1219,11 +1200,28 @@ static int
...
+                                "Cannot delete the x attribute");
+                break;
+            }
+            case 1: {
+                PyErr_SetString(PyExc_TypeError,
+                                "Cannot delete the y attribute");
+                break;
+            }
+            case 2: {
+                if (self->dim >= 3) {
```
Comment (ankith26) [inline src_c/math.c:1215]: Why is this check needed? I think in this path an error should be raised either way regardless of dim value

Comment (oddbookworm) [reply src_c/math.c:1215]: Differentiate between vector2 and vector3. This also guarantees the old behavior of going to badinternalcall if somehow someone manages to get a vec2 with a z property

Comment (ankith26) [reply src_c/math.c:1215]: The old code checked for 
```c
if (component >= self->dim) {
    PyErr_BadInternalCall();
}
```
Irrespective of the status of `value` but the changes do not preserve this behaviour.

Comment (oddbookworm) [reply src_c/math.c]: removed the check

**Review by Starbuck5 — APPROVED** [review]
Looks good to me, thanks!

File: src_c/math.c (hunk 1 above)
Comment (ankith26) [reply src_c/math.c]: Your latest commit didn't address what I was trying to say.

Consider a call where `value != NULL && component >= self->dim`. In this case the old code correctly raises an error but the new code doesn't. The `PyErr_BadInternalCall` check should happen independently of the `NULL` check outside the block and above it.

**Review by MyreMylar — APPROVED** [review]
OK, LGTM. The code makes sense and I can verify that the new tests fail before this PR is compiled and pass afterwards (tested locally).

**Pull request merged**